ipatests/test_integration/test_cert.py

```diff
@@ -385,40 +392,44 @@ def test_rekey_request_id(self, request_cert):
                                           '-I', self.request_id])
         assert self.request_id in result.stdout_text
 
-    def test_rekey_keytype_DSA(self):
+    def test_rekey_keytype_DSA(self, request_cert):
         """Test certmonger rekey command works fine
 
         Test is to check if -G (keytype) with DSA fails
 
         related: https://bugzilla.redhat.com/show_bug.cgi?id=1249165
         """
-        result = self.master.run_command(
-            [
-                'ipa-getcert', 'request',
-                '-f', os.path.join(paths.OPENSSL_CERTS_DIR, "test_dsa.pem"),
-                '-k', os.path.join(paths.OPENSSL_PRIVATE_DIR, "test_dsa.key"),
-                '-K', 'test/{}'.format(self.master.hostname),
-            ]
-        )
-        req_id = re.findall(r'\d+', result.stdout_text)
-        status = tasks.wait_for_request(self.master, req_id[0], 100)
+        status = tasks.wait_for_request(self.master, self.request_id, 100)
         assert status == "MONITORING"
 
         # rekey with RSA key type
         self.master.run_command(['getcert', 'rekey',
-                                 '-i', req_id[0],
+                                 '-i', self.request_id,
                                  '-g', '3072',
                                  '-G', 'DSA'])
         time.sleep(100)
+        # get all the request ids
+        result = self.master.run_command(
+            ['ls', '-l', paths.CERTMONGER_REQUESTS_DIR]
+        )
+        req_ids = re.findall(r'\d{14}', result.stdout_text)
+
+        # get the cert request file in certmonger request file
+        # by comparing list of cert request file with the list
+        # which we fetched before requesting new cert i.e self.req_ids
+        req_file = [req_id for req_id in req_ids if req_id not in self.req_ids]
```

Comment:
You can simplify the logic. For instance, when looking for the file name for request ID=20210322130730:
```
# find /var/lib/certmonger/requests/ -type f -exec grep -l "^id=20210322130730$" \{} \;
/var/lib/certmonger/requests/20210322130730
```
or
```
# grep -rl "^id=20210322130730$" /var/lib/certmonger/requests/
/var/lib/certmonger/requests/20210322130730
```
No need to store the list of cert request files before/after the getcert request.